x-pack/legacy/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/create_analytics_form/create_analytics_form.tsx

```diff
@@ -116,16 +125,50 @@ export const CreateAnalyticsForm: FC<CreateAnalyticsFormProps> = ({ actions, sta
     }
   };
 
-  // Regression supports numeric fields. Classification supports numeric, boolean, text, keyword and ip.
-  const shouldAddFieldOption = (field: Field) => {
+  // Regression and Classification support numeric, boolean, text, keyword, and ip types
+  // Outlier detection supports numeric or boolean types
+  const shouldAddAsAnalyzedFieldsOption = (field: Field) => {
```

Comment:
As discussed, we should try and populate the list of available fields by using the explain API.

Reply:
Updated here 24bdeb3e4dbf7b711cb9f257118c356b6baa8c38